subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/annotation/validation/HasSideEffects.kt

```diff
@@ -0,0 +1,9 @@
+package com.avito.android.runner.annotation.validation
+
+import org.junit.rules.TestRule
+
+/**
+ * Marker for non-hermetic test rules
+ * They should be used only in e2e tests
+ */
+interface HasSideEffects : TestRule
```

Comment:
I will add this marker to `MessengerConnectionRule` and `ResourcesRule`.

Reply:
It looks unscalable. How do you think we will control that each Rule must be marked with that interface?

Reply:
If we used subtyping for tests we could make rules like an extension for concrete types of tests

Reply:
I don't see how to detect side effects in general. You could write any logic in a custom test rule.
I see this as a helper for test rules, literally a marker to describe and define a contract.

The mocking of the network is happening in a test application on DI level. The decision is made based on applied test rules.
I see another option - mock network on a lower level, but this might be a huge change.

Let's discuss it by voice.

Reply:
Removed this interface. Now, this is the responsibility of a test rule.